crypto/provider_core.c

```diff
@@ -1672,6 +1674,30 @@ static int core_pop_error_to_mark(const OSSL_CORE_HANDLE *handle)
 {
     return ERR_pop_to_mark();
 }
+
+static int core_obj_add_sigid(const OSSL_CORE_HANDLE *prov,
+                              const char *sign_name, const char *digest_name,
+                              const char *pkey_name)
+{
+    int sign_nid = OBJ_txt2nid(sign_name);
+    int digest_nid = OBJ_txt2nid(digest_name);
+    int pkey_nid = OBJ_txt2nid(pkey_name);
+
+    if (sign_nid == NID_undef
+            || digest_nid == NID_undef
+            || pkey_nid == NID_undef)
+        return 0;
+
+    return OBJ_add_sigid(sign_nid, digest_nid, pkey_nid);
```

Comment:
OBJ_add_sigid() does not check whether there is already such triplet registered. That means we might get the same triplets registered multiple times if a provider is loaded/unloaded multiple times. Do we care or should we do a similar check as done below for OBJ_create?

Reply:
Yes. I think we care. Fixed.